src/Offer/ProcessManagers/RelatedDocumentProjectedToJSONLDDispatcher.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Offer\ProcessManagers;
+
+use Broadway\Domain\DomainEventStream;
+use Broadway\Domain\DomainMessage;
+use Broadway\Domain\Metadata;
+use Broadway\EventHandling\EventBus;
+use Broadway\EventHandling\EventListener;
+use CultuurNet\UDB3\Event\Events\EventProjectedToJSONLD;
+use CultuurNet\UDB3\Event\ReadModel\Relations\RepositoryInterface as EventRelationsRepository;
+use CultuurNet\UDB3\Iri\IriGeneratorInterface;
+use CultuurNet\UDB3\Organizer\OrganizerProjectedToJSONLD;
+use CultuurNet\UDB3\Place\Events\PlaceProjectedToJSONLD;
+use CultuurNet\UDB3\Place\ReadModel\Relations\RepositoryInterface as PlaceRelationsRepository;
+
+final class RelatedDocumentProjectedToJSONLDDispatcher implements EventListener
+{
+    private const METADATA_KEY = 'embeds_updated';
+
+    private EventBus $eventBus;
+    private EventRelationsRepository $eventRelationsRepository;
+    private PlaceRelationsRepository $placeRelationsRepository;
+    private IriGeneratorInterface $eventIriGenerator;
+    private IriGeneratorInterface $placeIriGenerator;
+
+    public function __construct(
+        EventBus $eventBus,
+        EventRelationsRepository $eventRelationsRepository,
+        PlaceRelationsRepository $placeRelationsRepository,
+        IriGeneratorInterface $eventIriGenerator,
+        IriGeneratorInterface $placeIriGenerator
+    ) {
+        $this->eventBus = $eventBus;
+        $this->eventRelationsRepository = $eventRelationsRepository;
+        $this->placeRelationsRepository = $placeRelationsRepository;
+        $this->eventIriGenerator = $eventIriGenerator;
+        $this->placeIriGenerator = $placeIriGenerator;
+    }
+
+    public static function hasDispatchedMessage(DomainMessage $domainMessage): bool
+    {
+        return $domainMessage->getMetadata()->get(self::METADATA_KEY) === true;
+    }
+
+    /**
+     * @uses handlePlaceProjectedToJSONLD
+     * @uses handleOrganizerProjectedToJSONLD
+     */
+    public function handle(DomainMessage $domainMessage): void
+    {
+        $payload = $domainMessage->getPayload();
+        $payloadClassName = get_class($payload);
+
+        $handlers = [
+            PlaceProjectedToJSONLD::class => 'handlePlaceProjectedToJSONLD',
+            OrganizerProjectedToJSONLD::class => 'handleOrganizerProjectedToJSONLD',
+        ];
+
+        $handler = $handlers[$payloadClassName] ?? null;
+        if ($handler) {
+            $this->{$handler}($payload);
+        }
+    }
+
+    private function handlePlaceProjectedToJSONLD(PlaceProjectedToJSONLD $placeProjectedToJSONLD): void
+    {
+        $placeId = $placeProjectedToJSONLD->getItemId();
+
+        // In theory the event relations repository should only return unique event ids for a given place id, but to be
+        // safe pass it through array_unique() so we don't needlessly dispatch duplicate messages.
+        $eventIds = $this->eventRelationsRepository->getEventsLocatedAtPlace($placeId);
```

Comment:
The table `event_relations` has a primary key on `event` so it should be unique. Although it doesn't harm to call `array_unique`

Reply:
Yes but that is only one implementation, and the implementation could in theory change. And like you said it doesn't harm :)